### ISSUE TITLE
Can't set keyword arguments sample_weight and init_model while using a custom loss

### ISSUE THREAD

**Issue by tiagofrepereira2012** [catboost/catboost#2670]
Problem:

While using custom loss it is not possible to set the both keyword arguments `sample_weight` and `init_model` (`CatboostRegressor.fit` method).
Weights are all set to 1.0

Follow below a snippet to reproduce the issue. You can see that the weights that are provided to `MultiRMSEObjective.calc_ders_multi` are all set to 1 instead of using the `sample_weight` provided.

```python
import numpy as np
from catboost import CatBoostRegressor, MultiTargetCustomMetric, MultiTargetCustomObjective


class MultiRMSEObjective(MultiTargetCustomObjective):
    """
    Custom RMSE objective for multi-target regression.

    This is a copy from their documentation. Using this as a reference to create a
    custom objective for multi-target regression.
    """

    def calc_ders_multi(self, approx, target, weight):
        """
        Computes first derivative and Hessian matrix of the loss function
        with respect to the predicted value for each dimension.

        Parameters
        ----------
        approxes : indexed container of floats
            Predictions for each dimension of single object.

        targets : single expected value
            True label.

        weight : float, optional (default=None)
            Instance weight.

        Returns
        -------
            der1 : list-like object of float
            der2 : list of lists of float

        """

        print(weight) # HERE WEIGHTS ARE ALWAYS ONE

        w = weight if weight is not None else 1.0
        der1 = [(target[i] - approx[i]) * w for i in range(len(approx))]
        der2 = [-w for i in range(len(approx))]

        # Convert to numpy arrays, float
        der1 = np.array(der1, dtype=np.float64)
        der2 = np.array(der2, dtype=np.float64)

        return (der1, der2)


class MultiRMSEMetric(MultiTargetCustomMetric):
    def get_final_error(self, error, weight):
        return np.sqrt(error / (weight + 1e-38))

    def is_max_optimal(self):
        return False

    def evaluate(self, approxes, target, weight):
        assert len(target) == len(approxes)
        assert len(target[0]) == len(approxes[0])

        error_sum = 0.0
        weight_sum = 0.0

        for i in range(len(approxes[0])):
            w = 1.0 if weight is None else weight[i]
            weight_sum += w
            for d in range(len(approxes)):
                error_sum += w * ((approxes[d][i] - target[d][i]) ** 2)
        return error_sum, weight_sum


# Usage Example
# Create a sample dataset
X_train = np.random.rand(100, 10)
y_train = np.random.rand(100, 3)  # Assuming 3 target variables
weights = np.random.rand(100)

# Setting up a dummy model that learns almost nothing
dummy_model = CatBoostRegressor(
    loss_function="MultiRMSE",
    leaf_estimation_method="Gradient",
    learning_rate=1e-6,
    boost_from_average=True,
    iterations=1,
).fit(X_train, y_train)


model_2 = CatBoostRegressor(
    loss_function=MultiRMSEObjective(),
    eval_metric=MultiRMSEMetric(),
    leaf_estimation_method="Gradient",
    bootstrap_type="Bayesian",
    # leaf_estimation_iterations=1,
)

model_2.fit(X_train, y_train, sample_weight=weights, init_model=dummy_model)

```

catboost version: 1.2.5
